internal/eventstore/handler/v2/handler.go

```diff
@@ -471,11 +471,11 @@ func (h *Handler) processEvents(ctx context.Context, config *triggerConfig) (add
 
 	txCtx := ctx
 	if h.txDuration > 0 {
-		var cancel func()
-		ctx, cancel = context.WithTimeout(ctx, h.txDuration)
-		defer cancel()
+		var cancel, cancelTx func()
 		// add 100ms to store current state if iteration takes too long
-		txCtx, cancel = context.WithTimeout(ctx, h.txDuration+100*time.Millisecond)
+		txCtx, cancelTx = context.WithTimeout(ctx, h.txDuration+100*time.Second)
+		defer cancelTx()
+		ctx, cancel = context.WithTimeout(ctx, h.txDuration)
```

Comment:
You replace the context which was used as parent for the txCtx with a child from the same ctx, is that planned?

Reply:
we use the initial ctx as base for both... since that could be canceled as well
and need additional timeouts (if configured)